PolyEngine/Core/Src/EnumUtils.hpp

```diff
@@ -68,6 +68,12 @@ namespace Poly {
 		EnumArray() : Data{} {}
 
 		//------------------------------------------------------------------------------
+		EnumArray(const std::initializer_list<std::pair<E, T>>& list)
```

Comment:
Please also provide implementation for `operator=`

Reply:
I only took a glimpse at it but what about an rvalue reference? 
To support std::move( T ) etc.

Reply:
If that is possible, that is. I don't know initializer_list that well.

Reply:
Thats actually a good idea. We would have to move each element separately though.

Reply:
How does one do that?
Variadic templates?

Reply:
Nananana, C++17-compliant Vuko to the rescue! 😜 
According to the standard, "_copying a std::initializer_list does not copy the underlying objects_" so we could just pass it by value. To add to that, the elements themselves are copy-initialized and then copied out of the underlying array. Unfortunately, moving them out is impossible since the array may be located in read-only memory.